### PR TITLE
Fix reflection deprecation

### PR DESCRIPTION
**Pull request by ekrich**

### CODE REVIEW

**Comment by ekrich** [comment]
This seems to run fine locally
`coursier % ./mill -i cli-tests.native-tests.test`

File: modules/launcher/src/main/scala/coursier/launcher/native/NativeBuilder.scala [hunk 1]
```diff
@@ -33,7 +33,7 @@ object NativeBuilder {
     val loader = new URLClassLoader(files.map(_.toURI.toURL).toArray, parentLoader)
 
     val cls = loader.loadClass("coursier.launcher.NativeBuilderImpl")
-    cls.newInstance().asInstanceOf[NativeBuilder]
+    cls.getDeclaredConstructor().asInstanceOf[NativeBuilder]
```
Comment (tgodzik) [inline modules/launcher/src/main/scala/coursier/launcher/native/NativeBuilder.scala:36]:
```suggestion
    cls.getDeclaredConstructor().newInstance().asInstanceOf[NativeBuilder]
```
?

**Review by tgodzik — APPROVED** [review]
Looks good, though I wonder why this didn't pop up in tests :thinking:

**Comment by ekrich** [comment]
Here is a little reference to this, different in Scala because we don't have checked exceptions - https://errorprone.info/bugpattern/ClassNewInstance

**Comment by ekrich** [comment]
I wondered about that too - I was too focused on the deprecation and that darn cast hid the bug I was trying to introduce.
I was running the following - I am new to Mill and Coursier too.
```sh
./mill -i cli-tests.native-tests.test
```

**Pull request merged**